iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/utils/CheckUtils.java

```diff
@@ -39,6 +39,14 @@ public static Predicate<Expression> isEqual(String targetString) {
     return expr -> TextUtils.matchesValue(expr, targetString::equals).isTrue();
   }
 
+  public static Predicate<Expression> contains(String targetString) {
+    return expr -> TextUtils.matchesValue(expr, containsPredicate(targetString)).isTrue();
+  }
+
+  private static Predicate<String> containsPredicate(String targetString) {
+    return predicate -> predicate.contains(targetString);
+  }
+
```

Comment:
```suggestion
  public static Predicate<Expression> contains(String targetString) {
    return expr -> TextUtils.matchesValue(expr, str -> str.contains(targetString)).isTrue();
  }
```
I would simplify it this way.